### DISCUSSION TITLE
Service Plan IDs

### DISCUSSION THREAD

**Discussion by jankrieg** [SAP/terraform-provider-btp#714] [Q&A] [answered]
Hi guys,

is there an easy way to get the service plan ids? Example: I'd like to create a service instance e.g. for the event mesh. 

Thanks,
Jan

**Comment by lechnerc77** [comment, marked as answer]
Hi @jankrieg ,

you can use the datasource [`btp_subaccount_service_plan`](https://registry.terraform.io/providers/SAP/btp/latest/docs/data-sources/subaccount_service_plan) to fetch the plan by name:

```terraform
data "btp_subaccount_service_plan" "by_name" {
  subaccount_id = "6aa64c2f-38c1-49a9-b2e8-cf9fea769b7f"
  name          = "free"
  offering_name = "alert-notification"
}
```

The data fetched from the datasource contains the ID of the plan that you can use to create a service instance via the resource [`btp_subaccount_service_instance`](https://registry.terraform.io/providers/SAP/btp/latest/docs/resources/subaccount_service_instance)

Is this what you are looking for?

**Comment by lechnerc77** [comment]
No respomnse for a month so closing this question
